### PR TITLE
Migrate to Spine 1.7.0

### PR DESCRIPTION
**Pull request by yuri-sergiichuk**
This PR migrates the example app to the latest released Spine v1.7.0 and resolves the breaking changes related to the migration.

As part of the PR, I have also updated the IDEA configs effectively copying them from the latest `config` module version (the commit https://github.com/SpineEventEngine/config/commit/c74b74f04e808224bcfb0ed2e7a4be925de5d273).

I have also addressed new errorprone warnings while I had to update dependencies in order to match ones from the new Spine libraries.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,57 +1,101 @@
-# Package Files #
-*.jar
-*.war
-*.ear
+#
```
Comment (alexander-yevsyukov) [inline .gitignore:1]: I don't think we need to have (c) statement in `.gitignore`

File: README.md [hunk 2]
```diff
@@ -1,10 +1,15 @@
...
 
 # Kanban Board example
 
 This example demonstrates:
-  * Multiple related Aggregates
-  * Orchestrating Aggregates using Process Managers.
-  * Building a Projection which is subscribed to both events and entity updates.
 
+* Multiple related Aggregate.
+
```
Comment (alexander-yevsyukov) [inline README.md:12]: Why this change?

**Review by alexander-yevsyukov — APPROVED** [review]
LGTM, with two comments.

File: .gitignore (hunk 1 above)
Comment (yuri-sergiichuk) [reply .gitignore:1]: Yep, I'll remove this one from here and I created the issue https://github.com/SpineEventEngine/config/issues/163 to refine where we'd like to have (c) and where not.

File: README.md (hunk 2 above)
Comment (yuri-sergiichuk) [reply README.md:12]: Just looks a bit spacier for my liking. But, I guess in this case it's better to have them condensed.
